### PR TITLE
Integration between thea & staking pallet

### PR DESCRIPTION
**Pull request by nakul1010**
Closes issue #610.

### CODE REVIEW

File: pallets/thea-staking/Cargo.toml [hunk 1]
```diff
@@ -25,9 +25,14 @@ sp-runtime = { default-features = false, workspace = true }
 sp-std = { default-features = false, workspace = true }
 sp-staking = { default-features = false, workspace = true }
 polkadex-primitives = { workspace = true, default-features = false }
+pallet-assets = { workspace = true, default-features = false }
+thea-primitives = {path = "../../thea-primitives", default-features = false}
+asset-handler = {path = "../asset-handler", default-features = false}
+chainbridge = {path = "../chainbridge", default-features = false}
```
Comment (serhii-temchenko) [inline pallets/thea-staking/Cargo.toml]: Please use common formatting with spaces between parenthesis.

File: pallets/thea-staking/Cargo.toml [hunk 2]
```diff
@@ -25,9 +25,14 @@ sp-runtime = { default-features = false, workspace = true }
...
 
 [dev-dependencies]
 sp-core = { default-features = false, workspace = true }
+thea = { path = "../thea", default-features = false}
```
Comment (serhii-temchenko) [inline pallets/thea-staking/Cargo.toml]: Same

File: pallets/thea-staking/src/lib.rs [hunk 3]
```diff
@@ -94,7 +98,10 @@ pub mod pallet {
 		type StakingDataPruneDelay: Get<SessionIndex>;
 
 		// TODO: @Faizal uncomment the code below for session change hook
-		//type SessionChangeNotifier: SessionChangeHook;
+		type SessionChangeNotifier: SessionChanged<
+			Network = Network,
+			OnSessionChange = OnSessionChange<Self::AccountId>,
+		>;
```
Comment (serhii-temchenko) [inline pallets/thea-staking/src/lib.rs]: I assume todo could be removed?

File: pallets/thea/Cargo.toml [hunk 4]
```diff
@@ -32,6 +32,7 @@ chainbridge = {path = "../chainbridge", default-features = false}
 asset-handler = {path = "../asset-handler", default-features = false}
 polkadex-primitives = { workspace = true, default-features = false }
 thea-primitives = {path = "../../thea-primitives", default-features = false}
+thea-staking = {path = "../thea-staking", default-features = false}
```
Comment (serhii-temchenko) [inline pallets/thea/Cargo.toml]: Please fix it here as well

File: thea-primitives/Cargo.toml [hunk 5]
```diff
@@ -25,5 +25,5 @@ std = [
     "frame-system/std",
     "parity-scale-codec/std",
     "scale-info/std",
-    "sp-core/std"
-]
+    "sp-core/std",
+]
```
Comment (serhii-temchenko) [inline thea-primitives/Cargo.toml]: Please add empty line at the end

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]

**Review by serhii-temchenko — APPROVED** [review]

**Review by Gauthamastro — APPROVED** [review]

**Pull request merged**